File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        res=0
        for i in s:
            if i-1 not in s:
                ct=1
                while i+1 in s:
                    ct+=1
                    i+=1
                res=max([res,ct]) 
        return res

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        for(int i:nums){
            st.insert(i);
        }
        int m=0,ct=0;
        for(int i:st){
            //ct=1;
            if(st.find(i-1)==st.end()){
                ct=1;
                while(st.find(i+1)!=st.end()){
                    ct++;
                    i++;
                }
                m=max(m,ct);
            }
        }
        return m; 
    }
};